### DISCUSSION TITLE
customtkinter.set_default_color_theme by event

### DISCUSSION THREAD

**Discussion by smokeflypaper** [TomSchimansky/CustomTkinter#449] [Q&A] [answered]
Is there is a way how to change color theme in the mainloop? 
I already created my own themes .json and i am trying to use that by choose from optionmenu. 
I can change the colors of buttons adn other widgets by my_button.configure(fg_color=my_color)
but i cant figure out how it work with styles. 
When i do: 
customtkinter.set_default_color_theme ("my_path_to\my_theme.json") it is working only out of the mainloop once when program is started but nothing happend in new_function(option_menu_event) inside the mainloop. 

I will appreciate all suggestions. 
p.e.a.c.e.

**Comment by Akascape** [comment, marked as answer]
@smokeflypaper Unfortunately, we can't change the whole theme in the mainloop. 

You can only switch between the light/dark attributes of one theme file at a time. 

See this discussion for more details: https://github.com/TomSchimansky/CustomTkinter/issues/295 

What you can do is to write the code that saves the theme name in a config file whenever the user change the theme options and then ask them to restart the program so that the new theme gets imported.

**Reply by smokeflypaper** [reply]
@Akascape thank you Sir! 
i know better now and i have the solution for my program too. 
I will make that config option for users..
